siliconcompiler/core.py

```diff
@@ -5176,67 +5206,131 @@ def _generate_testcase(self,
                 for _, key_step, key_index in self.schema._getvals(*key):
                     self.hash_files(*key, step=key_step, index=key_index)
 
-        manifest_path = os.path.join(issue_dir.name, 'manifest.json')
+        manifest_path = os.path.join(issue_dir.name, 'orig_manifest.json')
         self.write_manifest(manifest_path)
 
         flow = self.get('option', 'flow')
         tool, task = self._get_tool_task(step, index, flow=flow)
 
         # Set copy flags for _collect
-        for keypath in self.allkeys():
-            if 'default' in keypath:
-                continue
-
-            sctype = self.get(*keypath, field='type')
-            if 'file' not in sctype and 'dir' not in sctype:
-                continue
+        self.set('option', 'copyall', False)
 
+        def determine_copy(*keypath):
             copy = True
-            if keypath[0] == 'libraries':
+            if keypath[0] == 'library':
                 # only copy libraries if selected
                 if include_specific_libraries and keypath[1] in include_specific_libraries:
                     copy = True
                 else:
                     copy = include_libraries
+
+                copy &= determine_copy(*keypath[2:])
```

Comment:
Looks clean but using bitwise operators on booleans in Python makes me feel uneasy... might be best to do `copy = copy and determine_copy(...)`

Reply:
(using the boolean `and` also has a performance benefit since it'll short-circuit if copy is already false)

Reply:
Will do